src/main/java/org/prebid/server/bidder/improvedigital/ImprovedigitalBidder.java

```diff
@@ -71,16 +111,43 @@ public ImprovedigitalBidder(String endpointUrl, JacksonMapper mapper) {
         return bidsFromResponse(bidRequest, bidResponse);
     }
 
-    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+    private List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
         return bidResponse.getSeatbid().stream()
                 .filter(Objects::nonNull)
                 .map(SeatBid::getBid)
                 .filter(Objects::nonNull)
                 .flatMap(Collection::stream)
-                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur()))
+                .map(bid -> BidderBid.of(bidWithDealId(bid), getBidType(bid.getImpid(), bidRequest.getImp()),
+                        bidResponse.getCur()))
+                .filter(Objects::nonNull)
                 .collect(Collectors.toList());
     }
 
+    private Bid bidWithDealId(Bid bid) {
+        if (bid.getExt() != null) {
+            final ImprovedigitalBidExt improvedigitalBidExt;
+            try {
+                improvedigitalBidExt = mapper.mapper().treeToValue(bid.getExt(), ImprovedigitalBidExt.class);
+            } catch (JsonProcessingException e) {
+                throw new PreBidException(e.getMessage(), e);
+            }
+            final ImprovedigitalBidExtImprovedigital improve = improvedigitalBidExt.getImprovedigital();
+            if (improve == null) {
+                return bid;
+            }
+            // Populate dealId
+            String buyingType = improve.getBuyingType();
+            Integer lineItemId = improve.getLineItemId();
```

Comment:
Can be final

Reply:
Can we update lineitemId to be of String type and then instead of `lineItemId != null && lineItemId > 0` use `StringUtils.isNumeric(lineItemId)`